### PR TITLE
Run mypy as part of gha

### PR DESCRIPTION
**Pull request by ankitade**
Summary:
- Run mypy on all files on PRs. I tried doing it for files changes by PR but then it starts looking at the config files also :(
- Add some ignores based for iopath changes

Test plan:
- made a typing error in cnn_lstm -> type checkinng workflow fails https://github.com/facebookresearch/multimodal/runs/6068075411?check_suite_focus=true
- Fixed the error -> workflow passes on this PR https://github.com/facebookresearch/multimodal/runs/6068955875?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -26,3 +25,27 @@ jobs:
           output: ' '
       - name: Lint modified files
         run: pre-commit run --files ${{ steps.file_changes.outputs.files }}
+  type_checking:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+      - name: Setup conda env
```
Comment (langong347) [inline .github/workflows/lint.yaml:33]: N00b Q: do we need to install the library in order to run mypy? Is it possible to just check out the python interpreter, source code and run on the PR changes (like in linter)?

Comment (ankitade) [reply .github/workflows/lint.yaml:33]: it complains about imports if you don't do installation of torch or might happen with other dependencies (repros locally too)

File: mypy.ini [hunk 2]
```diff
@@ -24,3 +24,9 @@ ignore_missing_imports = True
 
 [mypy-torchvision.*]
 ignore_missing_imports = True
+
+[mypy-torchmultimodal.fb.*]
```
[the remark this answers is not in the document]
Comment (ankitade) [reply mypy.ini:28]: technically this is for all fb and next one for file_io

**Comment by langong347** [comment]
LGTM.

**Review by langong347 — APPROVED** [review]

**Pull request closed without merging**